lib/ddtrace/contrib/httparty/request_patch.rb

```diff
@@ -0,0 +1,137 @@
+require 'ddtrace/ext/net'
+require 'ddtrace/ext/distributed'
+require 'ddtrace/propagation/http_propagator'
+require 'ddtrace/contrib/rest_client/ext'
+
+module Datadog
+  module Contrib
+    module HTTParty
+      # HTTParty RequestPatch
+      module RequestPatch
+        def self.included(base)
+          base.send(:prepend, InstanceMethods)
+        end
+
+        # InstanceMethods - implementing instrumentation
+        module InstanceMethods
+          def uri
+            # uri is a method not a property in underlying class
+            # cache it so that it is only called once per call to 'perform', and then delete it
+            @dd_uri ||= super
+          end
+
+          def perform(&block)
+            return super unless tracer.enabled
```

Comment:
We do this a lot in other, older instrumentation, so this is fine, no change required. But reminded me that (although out of scope for this PR) we should have `Tracer#trace` check this instead.

To make that possible, we'd need no-op spans given the current implementation, something we've thought about doing a few times. @marcotc, if we implemented that `tracer.trace { |span| span.instrument { super } }` behavior we discussed, we could do this without no-ops, making lines like this obsolete (you'd just check if enabled and invoke the block if disabled instead of tracing.)